src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T extends object>(
+  items: T[],
+  key: keyof T,
+): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  for (const item of items) {
+    const keyValue = item[key];
+    const keyValueStr = String(keyValue);
```

Comment:
You are converting the key value to a string without checking if the key value is `undefined` or `null`. This could lead to unintentional grouping under the 'undefined' or 'null' string keys. If this behavior is not intended, you should handle these cases explicitly.